### ISSUE TITLE
Outputs samples for codec delay and final padding

### ISSUE THREAD

**Issue by sdroege** [rust-av/av-vorbis#1]
Currently using the Vorbis decoder via rust-av outputs some garbage at the beginning and end, which looks like the codec delay and final padding of the last frame.

It would be good if this could be dropped before passing the data to the caller.

**Comment by est31** [comment]
I'm not sure, usually they should be cut off? In the lewton library I have code to verify that my output is the very same as the libvorbis output: https://github.com/RustAudio/lewton/blob/master/dev/cmp/tests/vals.rs

**Comment by est31** [comment]
Oh I see that the comparison code actually discards the first packet... but theoretically it should be empty anyway: https://github.com/RustAudio/lewton/blob/3224e949ff2fbed3e17e6abd31ad6558a4867a4f/dev/cmp/src/lib.rs#L96

**Comment by lu-zero** [comment]
The fields to deliver this information exist, but it isn't used in the Context layer yet. If you want to beat me at adding it you are welcome :)

**Comment by sdroege** [comment]
How do they exist exactly? You mean internally and not at the `Decoder` level yet either?

**Comment by sdroege** [comment]
> Oh I see that the comparison code actually discards the first packet... but theoretically it should be empty anyway

It's also at the end, or are you currently filling the last packet completely always?

**Comment by lu-zero** [comment]
The fields are `convergence_window` and `delay` in `CodecParams`. The padding samples should be trimmed by setting the sample count in the packet to the correct value.

**Comment by sdroege** [comment]
Ah I see. But that currently can't be passed to the `Decoder`, which might make sense to be added.

However for this specific case, would you like the dropping of the samples be implemented at the `Context` layer or to the specific codec implementation itself?

**Comment by lu-zero** [comment]
Good point, that structure exists in that shape to be shared across. It could probably live at the Decoder level since those properties should be known by the decoder as well.

Since it is fundamental during the Seek scenario I postponed adding it.
